next/components/forms/widget-components/SelectField/SelectField.tsx

```diff
@@ -83,11 +88,25 @@ const SelectFieldComponent: ForwardRefRenderFunction<HTMLDivElement, SelectField
   )
 
   useEffect(() => {
-    if (!isDropdownOpened) {
-      setIsOutsideClickProgressing(false)
-    }
+    setIsClickedOutsideSelect(false)
+    setIsClickedOutsideDropdown(false)
+    setShouldCloseClick(false)
   }, [isDropdownOpened])
 
+  useEffect(() => {
+    if (isClickedOutsideDropdown && !shouldCloseClick && isClickedOutsideSelect) {
+      setIsDropdownOpened(false)
+    }
+  }, [isClickedOutsideDropdown, isClickedOutsideSelect])
+
+  useEffect(() => {
+    setIsDropdownOpened(false)
+  }, [shouldCloseClick])
+
+  useOnClickOutside(clickOutsideRef, () => {
+    setIsClickedOutsideSelect(true)
+  })
```

Comment:
this is really hard to read behaviour - can't it be simplified ? what are you trying to achieve with this ?